pkg/mx/handler.go

```diff
@@ -231,6 +252,56 @@ func (h *Handler) getPodsForMXJob(mxjob *v1alpha1.MXJob) ([]*v1.Pod, error) {
 	return cm.ClaimPods(podList(pods))
 }
 
+// getServicesForMXJob returns the set of services that this mxjob should manage.
+// It also reconciles ControllerRef by adopting/orphaning.
+// Note that the returned are live services from apiserver,
+// because sdk do not provide the capability to list from local cache for now.
+func (h *Handler) getServicesForMXJob(mxjob *v1alpha1.MXJob) ([]*v1.Service, error) {
+	// Create selector.
+	selector, err := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{
+		MatchLabels: genLabels(mxjob, ""),
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("couldn't convert Job selector: %v", err)
+	}
+	// List all services to include those that don't match the selector anymore
+	// but have a ControllerRef pointing to this controller.
+	services := v1.ServiceList{TypeMeta: metav1.TypeMeta{
+		Kind:       "Service",
+		APIVersion: "v1",
+	}}
+	err = sdk.List(mxjob.Namespace, &services, sdk.WithListOptions(&metav1.ListOptions{LabelSelector: labels.Everything().String()}))
```

Comment:
It seems like that all services in this namespace will be returned, did I miss something?

Reply:
yes, all will be handled by ref mgr

Reply:
Then how if there are multiple MXJobs in a same namespace?

Reply:
by selector?

Reply:
it need list all for adopting/orphaning

Reply:
shall we filter the services by selector?

Reply:
all -> ref-mgr + selector -> filtered(+adopted/orphaned)

Reply:
I see, in ClaimServices.